### ISSUE TITLE
`crates.io` bans users without warning

### ISSUE THREAD

**Issue by oovm** [rust-lang/team#1548]
Today I received a huge number of CI failure emails. 

After investigation, this is because `crates.io` has banned my account and all API_TOKENs.

There was no sign, no email telling me about this, and no channel to find out why I was banned.

This behavior has caused serious economic losses to my company and my team, and the entire `GitOps` has completely collapsed.

As a team leader, my team members were unable to submit versions after the `API_TOKEN` was banned, and development was completely stagnant.

---

As an active rust developer, I am very disappointed with the rust crates team. You are like a makeshift team and have never considered commercial operation.

I think building a business landscape on crates is like a joke. There is no warning email at all, and there is no communication at all.

**You can send an email a week in advance to point out the problem, and set a deadline to correct it, and take action after the deadline.**

Now my boss asked me to estimate the time it would take to completely change to a local build flow, and I have to submit an incident report, all of which makes me very angry.

The accident level is still under evaluation, it may be T1 or T0.

Bad day, this is not fun.

**Comment by Kobzol** [comment]
Hi, this is not the correct repository for posting an issue like this (it is a mostly internal repo designed to manage the members of the Rust Project). You should post this either to https://github.com/rust-lang/crates.io/issues or to https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io.

**Comment by oovm** [comment]
Please don't mark as won't fix, that would reinforce my stereotype of rust-teams being arrogant and authoritarian.

**This is a dictatorial bug of the rust team, which undoubtedly needs to be fixed.**

So far, no one has given me a reasonable explanation for the ban, and no one has apologized to me, so I refuse to deal with this matter privately, and I ask for a public handling process.

**My goal is to establish a fair and open system**, not to give people the opportunity to exercise dictatorial power.

---

1. **For abnormal frequency of use of API_TOKENs, it should be detected by automated strategies, temporarily block the use for 24 hours, send a warning email, and require an explanation within a time limit.**

2. **Deleting crates can be proposed by any rust user, but must be agreed by two or more crates team members.**

**Then enter a 7-day public notice period, and send a warning email. If there is no objection after the expiration, it will be automatically deleted.**

3. **Reports of spam accounts can be made by any rust user, but must be agreed upon by two or more crates team members (anonymously).**

**The reasons for the ban need to be publicly recorded and permanently archived in a public repository similar to a [DMCA request](https://github.com/github/dmca). The reasons cannot be general nonsense, but must be accurate to the specific repo.**

**Then enter a 7-day public notice period, send a warning email, and give an explanation within a time limit.**

**If no reasonable explanation is given, the ban will be implemented.**

4. **The new ToS requires users to display and agree to it the next time they log in to `crates.io`.**

---

Since no one explained the reason for the ban, I can only speculate that it is
1. Too many API_TOKEN calls
2. Too many REPOs
3. Excessive name occupation.
4. Other ToS I have never seen or agreed to

I started using rust about 7 years ago. For a long time, crates did not have an org mechanism (so call `teams` on `crates.io`), so it was common to apply for multiple API_TOKENs and deploy them to the CI of different projects.

I have been to many companies, and I did not revoke the previous API_TOKEN. This is my mistake. I apologize for this.

Because the team is used to using `docs.rs` to browse the latest documents, I didn't know who the new team leader was when I left, so I couldn't transfer, and there was no team mechanism at that time.

For this reason, I will contact the former technical team, set up teams, and transfer some crates under my name.

---

The community has called for namespaces many times, but the crates team has considered many reasons and cannot implement this capability, which I understand.

The community spontaneously created the naming practice of project-xxx-xxx to indicate namespaces. Many well-known projects such as tree-sitter use this practice.

I also use this rule to create plugins for my projects, which may lead to an unusual number of projects.

I don't have a good solution for this, because it's the accepted way of doing things.

---

Regarding the name usurpation issue, I did write crates for many algorithms for sharing purposes, and the quality of the algorithms may not be so top-notch.

But I did not usurp any trademarked names, any well-known command line projects, or any well-known open source projects.

I have not blackmailed anyone, on the contrary I have actively responded to reasonable demands.

For example, in the [convex](https://crates.io/crates/convex/versions) project, after @jamescowling contacted me to buy it, I did not make any financial demands. I was very happy that someone could provide better value. 

After listening to their reasonable request, I subsequently migrated my algorithm to the [shapes project](https://docs.rs/shape-core/latest/shape_core/) and gave the ownership to @jamescowling the next day.

---

For the above reasons, I believe that crates teams have not established a correct democratic decision-making mechanism, and private communication can no longer solve the problem, so I seek the intervention and help of rust-teams.

Please close this issue after the correct system is established, rather than won't fix it. Thank you for your understanding!

**Comment by Kobzol** [comment]
Hi, I closed this issue because it is not in any way relevant to this repo. It is like you were disputing a GitHub DMCA takedown request, and then created a GitHub issue in the https://github.com/actions/checkout repository - it would not belong there.

Regarding your ban, I asked the crates.io team and they said that you should have received an e-mail notification with explanation of the ban, to which you can react.

If you want to discuss the crates.io policies, I would suggest to use the already linked Zulip channel (https://rust-lang.zulipchat.com/#narrow/stream/318791-t-crates-io).

Once again, this repository is not the correct place to discuss these matters.

**Comment by oovm** [comment]
The email associated with my github account did not receive any emails about this incident.
The command line gave a general prompt that the account violated the ToS and was permanently banned.
Therefore, I refused to communicate privately, and I thought my request was not responded to reasonably.

**Comment by Kobzol** [comment]
Please continue the discussion on Zulip, and ideally follow the suggestion of the crates.io team, to contact them at `help@crates.io`. This is the policy set by the crates.io team, if you want to get your account unlocked, it seems reasonable to follow it. You can discuss or comment on the procedure publicly in any way you like (but again, **it is not relevant here**) - I would suggest you to open a new topic in the `t-crates-io` Zulip channel.

**Comment by oovm** [comment]
I think you got one thing wrong. I reported this to the teams because I was seeking justice, not to unblock my account.

The incident has already happened, and I was charged with a T1 (internal classification, known as P1 in the public domain) incident. My team has dealt with the adverse effects of this incident and built a new GitOps. Whether the account is unblocked or not has no meaning to me.

Therefore I refuse to follow the authoritarian and wrong private communication methods.

My goal is to build an open, fair, equitable, and democratic crates system, but you have been perfunctory again and again.
I am very disappointed with the entire rust-teams.
